### PR TITLE
minor: hide param inlay hint when argument is fn-like macro with similar name

### LINKED ISSUE #12486: Hide param name inlay hint when argument is a fn-like macro with similar name
This is already done for regular identifiers. It would be nice if it were also done with fn-like macros. For example, in the image below, the `expect:` inlay-hint is unnecessary:
image

I would like to work on this issue.
If I read the source code correctly, it seems like a simple addition to the following function will suffice.
https://github.com/rust-lang/rust-analyzer/blob/66a842124b52cb005e9dee4605df161badb295e0/crates/ide/src/inlay_hints.rs#L1089-L1105

### PR DESCRIPTION
**Pull request by lukechu10**
Closes #12486

### CODE REVIEW

**Comment by lukechu10** [comment]
Hmm why is CI failing?

File: crates/ide/src/inlay_hints.rs [hunk 1]
```diff
@@ -1496,6 +1499,11 @@ fn main() {
     let param2 = 0;
     foo(param2);
       //^^^^^^ param
+
```
Comment (lnicola) [inline crates/ide/src/inlay_hints.rs:1502]: Some whitespace here.

Comment (lukechu10) [reply crates/ide/src/inlay_hints.rs]: Done!

**Comment by lnicola** [comment]
Thanks!

@bors r+

**Pull request merged**